engine/runtime/src/memories/noncontiguous_memory.rs

```diff
@@ -90,14 +90,14 @@ impl LockedMemory for NonContiguousMemory {
     /// Unlocks the memory and returns an unlocked Buffer
     /// To retrieve secret value you xor the hash contained in shard1 with value in shard2
     fn unlock(&self) -> Result<Buffer<u8>, MemoryError> {
-        let (data1, data2) = self.get_shards_data()?;
-        let data1 = &blake2b::Blake2b256::digest(data1);
-        let reconstructed_data = xor(data1, &data2, NC_DATA_SIZE);
+        let (r, mut m) = self.get_shards_data()?;
```

Comment:
more descriptive variables?

Reply:
We've discussed this. See https://github.com/iotaledger/stronghold.rs/pull/474#discussion_r1201537073. Do you have suggestions?

Reply:
No, because I don't know what this code does (because the variables aren't descriptive)

Reply:
So we're just not gonna change this?